### PR TITLE
fix(core): resolve some core no-restricted-syntax lint error

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Resolve some core no-restricted-syntax lint errors.
Comment added for those hard-to-bypass issues. 

## Testing

test locally
@logto-io/eng

### CODE REVIEW

File: packages/core/src/middleware/koa-slonik-error-handler.ts [hunk 1]
```diff
@@ -39,12 +39,16 @@ export default function koaSlonikErrorHandler<StateT, ContextT>(): Middleware<St
         case InsertionError:
           throw new RequestError({
             code: 'entity.create_failed',
+            // Error constructor does not assert the instance type
+            // eslint-disable-next-line no-restricted-syntax
             name: (error as InsertionError<SchemaLike>).schema.tableSingular,
           });
```
Comment (gao-sun) [inline packages/core/src/middleware/koa-slonik-error-handler.ts:45]: how about we just simply use `if (error instanceof Foo)` here?

**Review by gao-sun — APPROVED** [review]
coool

File: packages/core/src/middleware/koa-slonik-error-handler.ts (hunk 1 above)
Comment (simeng-li) [reply packages/core/src/middleware/koa-slonik-error-handler.ts:45]: Tried, `instanceof Class` does not accept a Generic type. like `instanceof InsertionError<SchemaLike>`. Without it, `error.schema` is the type of any, have to use as to assert a Generic type anyway.

Comment (gao-sun) [reply packages/core/src/middleware/koa-slonik-error-handler.ts:45]: hold on, let me try something

Comment (gao-sun) [reply packages/core/src/middleware/koa-slonik-error-handler.ts:45]: image

then let's use if + as, which looks more clear

**Review by gao-sun — APPROVED** [review]

File: packages/core/src/middleware/koa-slonik-error-handler.ts (hunk 1 above)
Comment (simeng-li) [reply packages/core/src/middleware/koa-slonik-error-handler.ts]: updated. Since quite straightforward. address this conversation directly.

**Pull request merged**